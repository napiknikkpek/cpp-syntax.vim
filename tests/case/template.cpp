foo->template get<int>();
Foo<int> f;
Foo<Bar> f;
Foo<3> f;
foo::Bar<int> f;
foo::Bar<A1> f;
foo::Bar<3> f;
foo<Bar, int, 3, N>(x, 3, N);
foo::bar<Bar, int, 3, N>(x, 3, N);
foo<T, bar<X>, bar<X, Y>> a;
foo<T, bar<X>, bar<X, Y>>();
foo<T, decltype(a > b), T>();
foo<decltype(a + bar<T>{} - zoo<bar<T>>{})>();

std::function<void(argument1&, const argument2&, decltype(a 
      > b), const argument4&)> updated;

template <typename T, int N, typename... Args>
void foo();
