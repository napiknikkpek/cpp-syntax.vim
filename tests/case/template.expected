xxx..aaaaaaaa.fff.zzz....
ttt.zzz..x.
ttt.xxx..x.
ttt.c..x.
ttt..ttt.zzz..x.
ttt..ttt.xx..x.
ttt..ttt.c..x.
fff.xxx..zzz..c..x..x..c..x..
tttt..fff.xxx..zzz..c..x..x..c..x..
